src/api/libcellml/logger.h

```diff
@@ -73,7 +126,51 @@ class LIBCELLML_EXPORT Logger
      *
      * @return A reference to the error at the given index on success, @c nullptr otherwise.
      */
-    ErrorPtr error(size_t index) const;
+    IssuePtr error(size_t index) const;
+
+    /**
+     * @brief Get the number of issues with level of WARNING.
+     *
+     * Return the number of warnings currently stored in the logger.
+     *
+     * @return The number of warnings.
+     */
+    size_t warningCount() const;
```

Comment:
isn't this the wrong way to be doing these? Surely if I add a new level to the `Level` enum I shouldn't also need to create the count methods. And the `error()`, `issue()`, and `hint()` methods are duplicating the trivially equivalent `issue(Level)` ?

Reply:
Are we really expecting to add new levels in the future? Personally, I am fine with that. At the end of the day, those are just convenience methods in the same way that we provide convenience methods for other aspects of libCellML.

Reply:
I only included these methods to make the code easier to read.  Using enums means we have statements like `issueCount(libcellml::Issue::Level::WARNING)` when `warningCount()` is a little nicer.  But as you say, it's a very trivial overload ...

Reply:
I think this is OK as a convenience method adding new ones if other `Level`'s are added is a simple task.